src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,537 @@
+
+        /* mpi reduce */
+        using namespace lambda;
+        (*this->allGPU_reduce)(hBufTotal, hBufLeftParsHistogram, _1 + _2);
+        if(!this->allGPU_reduce->root())
+            return;
+
+        splash::SerialDataCollector hdf5DataFile(1);
+        splash::DataCollector::FileCreationAttr fAttr;
+
+        splash::DataCollector::initFileCreationAttr(fAttr);
+
+        std::stringstream filename;
+        filename << checkpointDirectory << "/" << prefix << currentStep;
+
+        hdf5DataFile.open(filename.str().c_str(), fAttr);
+
+        typename PICToSplash<float_X>::type SplashTypeX;
+
+        splash::Dimensions bufferSize(hBufTotal.size().x(),
+                                      hBufTotal.size().y(),
+                                      hBufTotal.size().z());
+
+        hdf5DataFile.write(currentStep,
+                           SplashTypeX,
+                           this->numBinsEnergy == 1 ? DIM2 : DIM3,
+                           splash::Selection(bufferSize),
+                           "detectorLeftPars",
+                           &(*hBufTotal.origin()));
+
+        hdf5DataFile.close();
+    }
+
+
+    void pluginLoad()
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        if(!(this->openingTheta > float_X(0.0) && this->openingTheta <= float_X(360.0)))
+        {
+            std::cerr << "[Plugin] [" << this->prefix
+                      << "] openingTheta has to be within (0, 360]."
+                      << std::endl;
+            return;
+        }
+        if(!(this->openingPhi > float_X(0.0) && this->openingPhi <= float_X(360.0)))
+        {
+            std::cerr << "[Plugin] [" << this->prefix
+                      << "] openingPhi has to be within (0, 360]."
+                      << std::endl;
+            return;
+        }
+
+        this->maxTheta = float_X(0.5) * this->openingTheta;
+        this->maxPhi = float_X(0.5) * this->openingPhi;
+        /* convert units */
+        const float_64 minEnergy_SI = this->minEnergy * UNITCONV_keV_to_Joule;
+        const float_64 maxEnergy_SI = this->maxEnergy * UNITCONV_keV_to_Joule;
+        this->minEnergy = minEnergy_SI / UNIT_ENERGY;
+        this->maxEnergy = maxEnergy_SI / UNIT_ENERGY;
+
+        /* allocate memory buffers */
+        this->dBufHistogram = new DBufHistogram(this->numBinsTheta, this->numBinsPhi, this->numBinsEnergy);
+        this->dBufLeftParsHistogram = new DBufHistogram(this->dBufHistogram->size());
+        this->hBufHistogram = new HBufHistogram(this->dBufHistogram->size());
+        this->hBufTotalHistogram = new HBufHistogram(this->dBufHistogram->size());
+
+        /* fill histogram for left particles with zero */
+        this->dBufLeftParsHistogram->assign(float_X(0.0));
+
+        /* create mpi reduce algorithm */
+        PMacc::GridController<simDim>& con = PMacc::Environment<simDim>::get().GridController();
+        using namespace PMacc::math;
```

Comment:
I am not a big fan of `using namespace` in the middle of a very long member function.

either define a new namespace and use that (`namespace pm = PMacc::math` ?) or do not do so at all. additionally, limit the scope of this namespace usage, e.g., use `{}` blocks or put the first suggestion on top of the method.